### PR TITLE
Add some basic tasks to help/workflow

### PR DESCRIPTION
**Pull request by mocobeta**
They might be worth adding to `help/workflow.txt`, and to me, it'd be great if we make this file a single entry point for every developer. e.g., we can just add a link to it from https://github.com/apache/lucene/blob/main/CONTRIBUTING.md instead of explaining each grade command to readers.

### CODE REVIEW

**Comment by mocobeta** [comment]
@dweiss would you mind taking a look at this?

File: help/workflow.txt [hunk 1]
```diff
@@ -25,11 +25,22 @@ Assemble a single module's JAR (here for lucene-core):
 gradlew -p lucene/core assemble
 ls lucene/core/build/libs
 
+Assemble all JARs:
```
[the remark this answers is not in the document]
Comment (mocobeta) [reply help/workflow.txt:28]: I copied these lines then modified it:
https://github.com/apache/lucene/blob/9b96a05f50a42f2a4c46fccbadd3057ba9b22933/help/workflow.txt#L24-L25

If "JAR" is too specific, how about "artifacts"?
```
Assemble all Lucene artifacts (JARs, and so on).
gradlew assemble
```

Gradle's documentation says "Translates Assembly language source files into object files." This would be correct but too general to me...
https://docs.gradle.org/current/dsl/org.gradle.language.assembler.tasks.Assemble.html

File: help/workflow.txt [hunk 2]
```diff
@@ -25,11 +25,22 @@ Assemble a single module's JAR (here for lucene-core):
...
+gradlew assemble
+
 Create all distributable packages, POMs, etc. and create a
 local maven repository for inspection:
 gradlew mavenLocal
 ls -R build/maven-local/
 
+Assemble Javdocs on a module:
```
[the remark this answers is not in the document]
Comment (mocobeta) [reply help/workflow.txt:36]: > This is a general question on where to explain how gradle works (-p command) and whether this document should provide ready-to-use commands doing things or an explanation how to assemble those commands from scratch. 

To me, a list of ready-to-use commands would be sufficient for this help document; here I'd give priority to brevity over detailed explanations. The usage of `-p` would be easily inferred from the examples, and I think developers who need more information about Gradle can always refer to the Gradle's documentation or other resources (just as I do).

[the remark this answers is not in the document]
Comment (mocobeta) [reply help/workflow.txt:36]: > Separately - do you ever assemble javadocs for a single module? What's the point of doing that?

Yes, sometimes - to see the changes on javadocs are fine in HTML (and it is convenient for lining in my usage). I added this line mainly for developers who want to contribute documentation.

[the remark this answers is not in the document]
Comment (dweiss) [reply help/workflow.txt:36]: ok, thanks.

File: help/workflow.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dweiss) [reply help/workflow.txt:28]: LGTM!

**Comment by gautamworah96** [comment]
I don't have any comments on this PR but a general +1 on the effort. I've sometimes found myself stumbling around the package searching for gradle commands that do what I want to do. Having a single file with some ready to use commands sounds perfect. I would also give a +1 for directly linking it from CONTRIBUTING.md. The easier we make it for contributors the better!

I sometimes also use the `-Ptests.iters=<some_large_num>` param for beasting out multiple runs of a single test to catch easy edge cases that I might have missed (this was another trick that I just stumbled upon through JIRA). Maybe we could add this to the workflow file as well?

File: help/workflow.txt (hunk 1 above)
Comment (mocobeta) [inline help/workflow.txt:28]:
```suggestion
Assemble all Lucene artifacts (JARs, and so on):
```

[the remark this answers is not in the document]
Comment (mocobeta) [reply help/workflow.txt]: Updated in https://github.com/apache/lucene/pull/811/commits/9c41951601d2b07deae84e8700c7342385141754. I didn't touch the existing description for the same command with `-p`.

**Comment by mocobeta** [comment]
@gautamworah96 thanks for your comments.

> I sometimes use the -Ptests.iters=<some_large_num> param for beasting out multiple runs of a single test to catch random edge cases that I might have missed (this was another trick that I just stumbled upon through JIRA). Maybe we could add this to the workflow file as well?

I found `help/tests.txt` (the source for `gradlew helpTests` task) has a dedicated section about "Reiteration", where `-Ptests.iters` option is explained. 
https://github.com/apache/lucene/blob/main/help/tests.txt#L87

`test` task has many parameters, I think it'd be better to encourage devs to refer to that file than augment `workflow.txt`? (You can see a pointer says `run "gradlew :helpTests" for more` in L12 in workflow.txt.)

**Comment by gautamworah96** [comment]
Hmm. I had not looked at the `help/tests.txt` file. +1 on not adding all the comprehensive options to this workflow file.
LGTM overall. 

I think we can still do some more to help new contributors but I have no specific action items in my head. I'll create separate JIRAs/PRs if I come up with something.

**Comment by mocobeta** [comment]
> I think we can still do some more to help new contributors but I have no specific action items in my head. I'll create separate JIRAs/PRs if I come up with something.

Sounds great, thanks!

**Comment by mocobeta** [comment]
@dweiss would you mind if I push it to main?

**Comment by dweiss** [comment]
Please, go ahead, @mocobeta. I'm on Easter holidays and with intermittent access to the internet.

**Comment by mocobeta** [comment]
Thanks and apologies for interrupting your holidays, I'll merge this soon.
